### PR TITLE
new resource: `azurerm_spring_cloud_dev_tool_portal`

### PR DESCRIPTION
**Pull request by ms-henglu**
```
=== RUN   TestAccSpringCloudDevToolPortal_basic
=== PAUSE TestAccSpringCloudDevToolPortal_basic
=== CONT  TestAccSpringCloudDevToolPortal_basic
=== RUN   TestAccSpringCloudDevToolPortal_requiresImport
=== PAUSE TestAccSpringCloudDevToolPortal_requiresImport
=== CONT  TestAccSpringCloudDevToolPortal_requiresImport
=== RUN   TestAccSpringCloudDevToolPortal_complete
=== PAUSE TestAccSpringCloudDevToolPortal_complete
=== CONT  TestAccSpringCloudDevToolPortal_complete
=== RUN   TestAccSpringCloudDevToolPortal_update
=== PAUSE TestAccSpringCloudDevToolPortal_update
=== CONT  TestAccSpringCloudDevToolPortal_update
--- PASS: TestAccSpringCloudDevToolPortal_basic (1213.60s)
--- PASS: TestAccSpringCloudDevToolPortal_complete (1294.68s)
--- PASS: TestAccSpringCloudDevToolPortal_requiresImport (1333.18s)
--- PASS: TestAccSpringCloudDevToolPortal_update (1446.04s)
PASS
```

### CODE REVIEW

**Review by yuwzho — APPROVED** [review]

File: internal/services/springcloud/spring_cloud_dev_tool_portal_resource.go [hunk 1]
```diff
@@ -0,0 +1,404 @@
...
+		},
+
+		"spring_cloud_service_id": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validate.SpringCloudServiceID,
+		},
+
+		"feature": {
```
Comment (katbyte) [inline internal/services/springcloud/spring_cloud_dev_tool_portal_resource.go:75]: i don't think we need to put these into a block called features? could we bring these up to the top level

File: internal/services/springcloud/spring_cloud_dev_tool_portal_resource.go [hunk 2]
```diff
@@ -0,0 +1,404 @@
...
+			Type:     pluginsdk.TypeList,
+			Optional: true,
+			Computed: true,
+			MaxItems: 1,
+			Elem: &pluginsdk.Resource{
+				Schema: map[string]*pluginsdk.Schema{
+					"application_accelerator": {
```
Comment (katbyte) [inline internal/services/springcloud/spring_cloud_dev_tool_portal_resource.go:82]:
```suggestion
					"application_accelerator_enabled": {
```

File: internal/services/springcloud/spring_cloud_dev_tool_portal_resource.go [hunk 3]
```diff
@@ -0,0 +1,404 @@
...
+				return err
+			}
+
+			var model SpringCloudDevToolPortalModel
+			if err := metadata.Decode(&model); err != nil {
+				return fmt.Errorf("decoding: %+v", err)
+			}
+
+			_, err = client.Get(ctx, id.ResourceGroup, id.SpringName, id.DevToolPortalName)
+			if err != nil {
```
Comment (katbyte) [inline internal/services/springcloud/spring_cloud_dev_tool_portal_resource.go:227]:
```suggestion
			if _, err = client.Get(ctx, id.ResourceGroup, id.SpringName, id.DevToolPortalName); err != nil {
```

File: internal/services/springcloud/spring_cloud_dev_tool_portal_resource.go [hunk 4]
```diff
@@ -0,0 +1,404 @@
...
+
+	return &res
+}
+
+func expandSpringCloudDevToolPortalSsoProperties(input []SsoModel) *appplatform.DevToolPortalSsoProperties {
+	if len(input) == 0 {
+		return nil
+	}
+	sso := input[0]
+	return &appplatform.DevToolPortalSsoProperties{
```
Comment (katbyte) [inline internal/services/springcloud/spring_cloud_dev_tool_portal_resource.go:349]:
```suggestion
	if len(input) == 0 {
		return nil
	}
	sso := input[0]
	
	return &appplatform.DevToolPortalSsoProperties{
```

File: internal/services/springcloud/spring_cloud_dev_tool_portal_resource.go [hunk 5]
```diff
@@ -0,0 +1,404 @@
...
+	}
+	metadataUrl := ""
+	if properties.MetadataURL != nil {
+		metadataUrl = *properties.MetadataURL
+	}
+	scopes := make([]string, 0)
+	if properties.Scopes != nil {
+		scopes = *properties.Scopes
+	}
+	return []SsoModel{
```
Comment (katbyte) [inline internal/services/springcloud/spring_cloud_dev_tool_portal_resource.go:396]:
```suggestion
	if properties == nil {
		return []SsoModel{}
	}
	
	clientId := ""
	if properties.ClientID != nil {
		clientId = *properties.ClientID
	}
	
	clientSecret := ""
	if len(model.Sso) != 0 {
		clientSecret = model.Sso[0].ClientSecret
	}
	
	metadataUrl := ""
	if properties.MetadataURL != nil {
		metadataUrl = *properties.MetadataURL
	}
	
	scopes := make([]string, 0)
	if properties.Scopes != nil {
		scopes = *properties.Scopes
	}
	
	return []SsoModel{
```

**Review by katbyte — CHANGES_REQUESTED** [review]

**Review by katbyte — APPROVED** [review]
LGTM 🌵

**Pull request merged**